src/mono/mono/utils/mono-log-common.c

```diff
@@ -110,7 +110,17 @@ mono_log_write_logfile (const char *log_domain, GLogLevelFlags level, mono_bool
 
 #ifdef HAVE_LOCALTIME_R
 		struct tm tod;
+#ifdef HOST_WASI
+		// In the WASI prototype, this is being compiled using WASI SDK which says that time() returns int64, but is
```

Comment:
Shouldn't libSystem.Native also be built using wasi ? Building some stuff using emscripten looks very fragile.

Reply:
Yes, it definitely should. That's also what the TODO comment on L117 of this file says :)